### DISCUSSION TITLE
ФРМО. В чем смысл ТВСП

### DISCUSSION THREAD

**Discussion by eugenekogan-spb** [fhir-ru/core#63] [Административная РГ]
В группе полей, описывающих здание, есть поле ТВСП - территориально выделенное структурное подразделение.

Текст:
> Описание ТВСП Допускается наличие только одного объекта. Для того, чтобы отвязать здание от ТВСП - необходимо прислать запрос без указания объекта, при этом проверка на наличие хотя бы одного здания в ТВСП остаётся.
> Если текущий объект building вложен в атрибут buildings объекта moTerritorialDepartId / moTerritorialDepart не заполнять этот атрибут

Вопрос. Пусть мы опишем все здания медицинских организаций и укажем расположение в них функциональных подразделений. Какая еще первичная информация потребуется Минздраву для расчетов по ТВСП?

Кроме того, есть отдельная группа признаков ТВСП:
<img width="304" alt="image" src="https://user-images.githubusercontent.com/62525329/146547433-94c21515-a89b-47ef-8c5a-0bc9f6101195.png">

Складывается впечатление, что этими признаками помечаются те самые функциональные группы подразделений, которые мы планировали вести с помощью OrganizationAffilation.

**Comment by owa99** [comment]
Практическое значение ТВСП имеет только одно - многие отчеты по выполнению показателей ЕГИСЗ считаются по ТВСП. Например, если у МО 3 ТВСП и нужен 100% показатель по отправке документа определенного типа в РЭМД, то надо из всех 3-х ТВСП такой документ отправить. По сути ТВСП это 1..n зданий, расположенных рядом территориально и имеющие разные литеры как правило.
В-общем, это чисто "отчетный" показатель, к которому не привязаны ресурсы FHIR.

**Reply by eugenekogan-spb** [reply]
Согласен что атрибут ТВСП выдуман для отчетов. Тем не менее, решая задачу о ведении реестра медицинских организаций через ресурсы FHIR, мы обязаны предложить пользователям модель для решения их отчетной задачи.

Гипотезы:
1) ТВСП - это флаг (бинарный атрибут), которым помечается ресурс Organization. ТВСП считается передавшим документ в РЭМД, если в РЭМД пришел хотя бы один документ , подписанный врачом, работающим в этой Organization или вложенных по PartOf. Пользователь произвольно (по своему решению) устанавливает флаг ТВСП у подразделений, имеющих самостоятельный Location
2) ТВСП - это флаг (бинарный атрибут), которым помечается ресурс Location. ТВСП считается передавшим документ в РЭМД, если в РЭМД пришел хотя бы один документ, подписанный врачом, работающим в Organization, в которую входит данный Location по связи ManagingOrganization. Пользователь произвольно (по своему решению) устанавливает этот флаг.
3) ТВСП - это ресурс OgranizationAffilation, который пользователь создает для группировки подразделений по признаку территориального размещений в здании или группе близко расположенных зданий. ТВСП считается передавшим документ в РЭМД, если в РЭМД пришел хотя бы один документ, подписанный врачом, работающим в Organization, входящей в состав данной Affilation.

**Reply by owa99** [reply]
Думаю, что точно не 1-й и не 3-й вариант. ТВСП - это поименованный список зданий МО, имеющий специализированные признаки. Поскольку ТВСП не имеет никаких явных идентификаторов, то его нет смысла хранить отдельно, вне ресурса Organization. Наверное, информацию о ТВСП можно хранить в Organization как сложный тип данных.
Со специализированными признаками ТВСП тоже не все гладко. 
Например, признак "Перинатальный центр" уже является **Видом** структурного подразделения:
![ПеринатЦентр](https://user-images.githubusercontent.com/19589199/147200906-28c4fa82-6e04-4995-bde2-fa019231aba4.png)
Зачем его дублировать на уровне ТВСП? 
Вообще, по справочникам, относящимся к структуре организации есть много вопросов.
Начнем с того, что структура ФРМО жестко 3-х уровневая:
![Структура ФРМО](https://user-images.githubusercontent.com/19589199/147202398-726391c0-b511-4fda-bc0a-04d780eed004.png)
То есть структура ФРМО не позволяет корректно вносить структурные подразделения даже по Приложению "В12/1".
Приведу пример на основе нашей МО (областная клиническая больница). 
В состав МО входит подразделение "Перинатальный центр", структура которого по приказу МЗ №1130н помимо стационарных отделений, включает амбулаторные подразделения: консультативно-диагностическое отделение со многими кабинетами и даже выездные акушерские и неонатологические бригады, которые оказывают медицинскую помощь вне медучереждения и по сути являются скоропомощными.
Однако, ФРМО хоть и позволяет в раздел "Структурные подразделения" внести вид СП "Перинатальный центр", но только с типом подразделения "стационарный" и сответственно, иметь в себе только отделения стационара. А клинико-диагностический центр приходится показывать как отдельное СП амбулаторного типа, без связи с  Перинатальным центром. 
То есть если коротко, СП по ФРМО может быть только одного типа, хотя подобных ограничений  в Приложении "В12/1" "Структурные подразделения медицинских организаций" от ЦНИИОИЗ я не увидел.
Есть и другие противоречия и непонятные моменты по структуре организации. Получается, у одной организации уже 3 структуры - одна реальная, другая по Приложению "В12/1" и третья по ЕГИСЗ? 
Думаю, что если мы будем пытаться делать ПОЛНУЮ проекцию условной совы "ЕГИСЗ" на условный глобус "FHIR", то только усложним ситуацию. И простого решения не будет
